msm/src/circuit_design/witness.rs

```diff
@@ -184,7 +193,37 @@ impl<
     > LookupCap<F, CIx, LT> for WitnessBuilderEnv<F, CIx, N_WIT, N_REL, N_DSEL, N_FSEL, LT>
 {
     fn lookup(&mut self, table_id: LT, value: Vec<<Self as ColAccessCap<F, CIx>>::Variable>) {
-        let value_ix = table_id.ix_by_value(&value);
+        let value_ix = if table_id.is_fixed() {
+            table_id
+                .ix_by_value(&value)
+                .expect("Could not resolve lookup for a fixed table")
+        } else {
+            // For runtime tables, we check if this value was already
```

Comment:
Is this necessarily the desired behaviour? I would have thought that the index would be 'fixed' for all our uses, given that we treat the tables effectively as arrays. That is to say, usually we'll be doing a lookup of the form `table_id + value_index * joint_combiner + value * joint_combiner^2`, and `value_index` determines exactly which value we're hoping to access.

Reply:
Hm. I'll think about it a bit later, but my current take is this:
1. Fixed tables have a fixed mapping `index -> value`, and we look up `table_id + value[0] * joint_combiner + value[1] * joint_combiner^2 + ...`
    * So `index` is not actually present in the lookup. It's only used in the code that constructs the lookup for counting multiplicities etc, since we request lookup by value, not by index. It's just an interface thing, we could do otherwise. But for now, if there are four value is the table, `555, 666, 777, 888`, we call it `lookup(column_777)` instead of `lookup_ix(column_777, 3)`. And so we record that lookup to 777 was done, and add `+1` to its multiplicity.
2. My current plan with runtime tables is to just mimic this behaviour. The only difference is that we don't know `index -> value` mapping in advance, so the code you're commenting on looks it up from a dynamically on-the-fly constructed map (resolver),  which is also maintained on-the-fly. Essentially we push elements to the current runtime table column, and when value from it is requested, we go to that value, and add `+1` to its multiplicity based on this index. It's very similar.

Reply:
I trust that you've thought about this, but I find it unlikely that this is actually the behaviour we want. I won't block though.

Reply:
@mrmr1993 Do I understand correctly that your suggestion wrt runtime tables specifically is to lookup `index || value_vec` instead of just `value_vec`? If yes, I probably understand the idea, but not the trade-offs: we seem to still need to maintain the reverse map `value -> index` while constructing the witness, so I'm not exactly seeing what's the advantage of this approach... what am I missing?